roomserver/internal/input/input_events.go

```diff
@@ -210,6 +219,34 @@ func (r *Inputer) processRoomEvent(
 	return event.EventID(), nil
 }
 
+func (r *Inputer) checkPowerLevels(
+	ctx context.Context, event gomatrixserverlib.Event,
+) (isRejected bool, err error) {
+	req := &api.QueryLatestEventsAndStateRequest{RoomID: event.RoomID()}
+	resp := &api.QueryLatestEventsAndStateResponse{}
+	if err := helpers.QueryLatestEventsAndState(ctx, r.DB, req, resp); err != nil {
+		logrus.WithError(err).Error("helpers.QueryLatestEventsAndState failed to get latest events and state")
+		return true, err
+	}
+	var creator string
+	for _, v := range resp.StateEvents {
+		if v.Type() == gomatrixserverlib.MRoomCreate {
+			creator = v.Sender()
+			break
+		}
+	}
+	if event.Sender() != creator && r.ServerName != event.Origin() {
```

Comment:
I don't think this check is doing the right thing - ultimately what we want to do is check if the event is sent by the same creator *only if* there isn't a power level event already in the room.

If there is an existing power level event then we don't need to do this, as the soft-fail check will capture it if the current state events don't allow the new event.